sigmastate/src/main/scala/sigmastate/eval/CostingDataContext.scala

```diff
@@ -746,18 +747,16 @@ case class CostingDataContext(
 
   @inline override def minerPubKey = _minerPubKey
 
-
-  private def findSelfBoxIndex: Int = {
-    var i = 0
-    while (i < inputs.length) {
-      if (inputs(i) eq selfBox) return i
-      i += 1
+  override def selfBoxIndex: Int = {
+    if (VersionContext.current.isEvaluateErgoTreeUsingJIT) {
```

Comment:
Again, as there are different versioning schemes used in 5.0 codebase (interpreter-based vs tree-based), please comment why tree based is used here

Reply:
changed to activation-based versioning